crates/astria-core/src/protocol/transaction/v1alpha1/action_group/tests.rs

```diff
@@ -0,0 +1,244 @@
+            bridge_address: address,
+            memo: String::new(),
+            rollup_block_number: 0,
+            rollup_withdrawal_event_id: String::new(),
+        }),
+        Action::ValidatorUpdate(ValidatorUpdate {
+            power: 100,
+            verification_key: VerificationKey::try_from([0; 32]).unwrap(),
+        }),
+        Action::Ics20Withdrawal(Ics20Withdrawal {
+            denom: asset.clone(),
+            destination_chain_address: String::new(),
+            return_address: address,
+            amount: 1_000_000u128,
+            memo: String::new(),
+            fee_asset: asset.clone(),
+            timeout_height: Height::new(1, 1).unwrap(),
+            timeout_time: 0,
+            source_channel: "channel-0".parse().unwrap(),
+            bridge_address: Some(address),
+            use_compat_address: false,
+        }),
+    ];
+
+    assert!(matches!(
+        Actions::try_from_list_of_actions(actions).unwrap().group(),
+        Some(ActionGroup::General)
+    ));
+}
+
+#[test]
+fn from_list_of_actions_sudo() {
+    let address: Address<_> = Address::builder()
+        .array([0; 20])
+        .prefix(ASTRIA_ADDRESS_PREFIX)
+        .try_build()
+        .unwrap();
+
+    let asset: Denom = "nria".parse().unwrap();
+    let actions = vec![
+        Action::FeeChange(FeeChangeAction {
+            fee_change: FeeChange::TransferBaseFee,
+            new_value: 100,
+        }),
+        Action::FeeAssetChange(FeeAssetChangeAction::Addition(asset)),
+        Action::IbcRelayerChange(IbcRelayerChangeAction::Addition(address)),
+    ];
+
+    assert!(matches!(
+        Actions::try_from_list_of_actions(actions).unwrap().group(),
+        Some(ActionGroup::Sudo)
+    ));
+}
+
+#[test]
+fn from_list_of_actions_unbundelable_sudo() {
+    let address: Address<_> = Address::builder()
+        .array([0; 20])
+        .prefix(ASTRIA_ADDRESS_PREFIX)
+        .try_build()
+        .unwrap();
+
+    let actions = vec![Action::SudoAddressChange(SudoAddressChangeAction {
+        new_address: address,
+    })];
+
+    assert!(matches!(
+        Actions::try_from_list_of_actions(actions).unwrap().group(),
+        Some(ActionGroup::UnbundleableSudo)
+    ));
+
+    let actions = vec![Action::IbcSudoChange(IbcSudoChangeAction {
+        new_address: address,
+    })];
+
+    assert!(matches!(
+        Actions::try_from_list_of_actions(actions).unwrap().group(),
+        Some(ActionGroup::UnbundleableSudo)
+    ));
+
+    let actions = vec![
+        Action::SudoAddressChange(SudoAddressChangeAction {
+            new_address: address,
+        }),
+        Action::SudoAddressChange(SudoAddressChangeAction {
+            new_address: address,
+        }),
+    ];
+
+    assert_eq!(
+        Actions::try_from_list_of_actions(actions)
+            .unwrap_err()
+            .to_string(),
+        "attempted to create bundle with non bundleable `ActionGroup` type: unbundleable sudo"
+    );
```

Comment:
```suggestion
    let error_kind = Actions::try_from_list_of_actions(actions).unwrap_err().0;
    assert!(
        matches!(error_kind, ErrorKind::NotBundleable { .. }),
        "expected ErrorKind::NotBundleable, got {error_kind:?}"
    );
```
Similar suggestion for following two tests also.